li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/metadataservice/MarioCommandCallbackImpl.java

```diff
@@ -31,19 +32,30 @@
   public void onReceivingCommand(Messages marioCommandMessage) {
     // Find a federate client callback that matches the given Mario command message type and execute it with arguments
     // included in the message.
+    LiKafkaFederatedClientType clientType = _federatedClient.getClientType();
 
     switch (marioCommandMessage.getMsgType()) {
       case RELOAD_CONFIG_REQUEST:
         ReloadConfigRequestMessages reloadConfigMsg = (ReloadConfigRequestMessages) marioCommandMessage;
 
-        if (_federatedClient.getClientType() == LiKafkaFederatedClientType.FEDERATED_PRODUCER) {
+        if (clientType == LiKafkaFederatedClientType.FEDERATED_PRODUCER) {
           // Call producer reload config method
           ((LiKafkaFederatedProducerImpl) _federatedClient).reloadConfig(reloadConfigMsg.getConfigs(), reloadConfigMsg.getCommandId());
         } else {
           // call consumer reload config method
           ((LiKafkaFederatedConsumerImpl) _federatedClient).reloadConfig(reloadConfigMsg.getConfigs(), reloadConfigMsg.getCommandId());
         }
         break;
+      case REGISTER_RESPONSE:
+        // Upon receiving register response from conductor, federated client would save the configs from the message and apply the
```

Comment:
corner case - this happens async, post construction, and the caller/constructor isnt blocked on this, right?
if so - is it possible that clients were created between construction and this point in time?

Reply:
Yes, the constructor is not blocked and the clients can be created in-between, therefore in applyBootupConfigFromConductor, it will check if clients are created, if yes, it will recreate them with the boot up configs.